# ./pullrev.sh r1861269
http://svn.apache.org/viewvc?view=revision&revision=r1861269

Allows "httpd -L" etc to work before httpd-init.service has run,
if mod_ssl is installed.

--- httpd-2.4.37/modules/ssl/ssl_engine_config.c
+++ httpd-2.4.37/modules/ssl/ssl_engine_config.c
@@ -904,8 +904,14 @@
 static const char *ssl_cmd_check_file(cmd_parms *parms,
                                       const char **file)
 {
-    const char *filepath = ap_server_root_relative(parms->pool, *file);
+    const char *filepath;
 
+    /* If only dumping the config, don't verify the paths */
+    if (ap_state_query(AP_SQ_RUN_MODE) == AP_SQ_RM_CONFIG_DUMP) {
+        return NULL;
+    }
+
+    filepath = ap_server_root_relative(parms->pool, *file);
     if (!filepath) {
         return apr_pstrcat(parms->pool, parms->cmd->name,
                            ": Invalid file path ", *file, NULL);
